ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/log/BootstrapDumpLogger.java

```diff
@@ -68,4 +68,9 @@ public void endLog(String lastReplId) {
     (new BootstrapDumpEnd(dbName, tableSeqNo, functionSeqNo, dumpDir, lastReplId))
             .log(LogTag.END);
   }
+
+  @Override
+  public void endLog(long totalCount) {
+    //Do Nothing
```

Comment:
Why not to provide default no-op version in base class itself and override only when required? Or even better, use generics for value top, that way only one version of endLog would be required?

Reply:
Atleast one endLog has to be present. If you provide a implementation in the base class, there maybe a derived class which doesn't override.

Reply:
What I was referring to is:
Option 1: If you use generics you don't need extra endLog method and you can still define endLog as abstract, just the signature will change to 
Option 2: If you want separate endLog for rangerLogger, you can provide default impl in base so that you don't need to touch the other unrelated classes.
So, for endLog if you see, polymorphism isn't working quite right. If you choose Option 1, I believe that should be fixed and base will behave like base for endLog operation as well.

Reply:
yes will use option 1